src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LoadedProject.cs

```diff
@@ -57,7 +66,34 @@ public LoadedProject(ProjectSystemProject projectSystemProject, SolutionServices
 
     private void FileChangedContext_FileChanged(object? sender, string filePath)
     {
-        NeedsReload?.Invoke(this, EventArgs.Empty);
+        // If the project file itself changed, we almost certainly need to reload the project.
+        if (string.Equals(filePath, _projectFilePath, StringComparison.OrdinalIgnoreCase))
```

Comment:
Are we gonna get .csproj file change events if we didn't include .csproj in CreateContext?